rules/explicit-length-check.js

```diff
@@ -161,6 +167,32 @@ function create(context) {
 				return;
 			}
 
+			const variable = findVariable(context.getScope(), lengthNode.object);
+			if (variable) {
+				const identifiers = getVariableIdentifiers(variable).filter(identifier => identifier !== lengthNode.object);
+				let expectedPrototype;
+				if (lengthNode.property.name === 'length') {
+					expectedPrototype = Array.prototype;
+				} else if (lengthNode.property.name === 'size') {
+					expectedPrototype = Set.prototype;
```

Comment:
This maybe hard than I thought, I don't think we mean to only check `Array#length` and `Set#size`, there are many builtin objects we want check,eg: `Map#size` `{File, Blob}.size`.

Reply:
Oh, I see. If we are just intending to check these built-in JS objects, shall I update this to check those prototypes too?

Reply:
Problem: too many builtins has `.length`.

I searched around, but didn't see a data with all those objects and their properties.


Image we have a list

```
File: ['size', 'method1', 'property1'],
Foo: ['size', 'method1', 'property1'],
Bar: ['noSize', 'method1', 'property1'],
```

We got all objects can has `.size` and all their methods and properties.

Reply:
Okay so you recommend closing this?

We should be able apply this strategy to other rules though which are limited to specific data structures, right?

Reply:
Les't make a list ourself. List `Map, Set Blob`'s property, and only check `.size` for now.

Reply:
And `fs.Stats` too.

Reply:
```js
Object.keys(Object.getOwnPropertyDescriptors(Set.prototype))
(10) ["constructor", "has", "add", "delete", "clear", "entries", "forEach", "size", "values", "keys"]

Object.keys(Object.getOwnPropertyDescriptors(Map.prototype))
(11) ["constructor", "get", "set", "has", "delete", "clear", "entries", "forEach", "keys", "size", "values"]

Object.keys(Object.getOwnPropertyDescriptors(Blob.prototype))
(7) ["size", "type", "arrayBuffer", "slice", "stream", "text", "constructor"]

Object.keys(Object.getOwnPropertyDescriptors(File.prototype))
(5) ["name", "lastModified", "lastModifiedDate", "webkitRelativePath", "constructor"]

Object.keys(Object.getOwnPropertyDescriptors(require('fs').Stats.prototype))
[ 'constructor', 'isFile', '_checkModeProperty' ]
```